lib/src/tree.dart

```diff
@@ -13,6 +13,9 @@ class Identifier extends TreeNode {
 
   Identifier(this.name, SourceSpan span) : super(span);
 
+  @override
+  SourceSpan get span => super.span!;
```

Comment:
I tried also to use `as SourceSpan` so that it could remain nullable in opt-out mode, but it does look like there are intentionally ways for null to get here so I'll have to make them nullable across the board.